### DISCUSSION TITLE
Program icons are not visible

### DISCUSSION THREAD

**Discussion by mintluv77** [InsertX2k/temp_cleaner_gui#15] [Q&A] [answered]
Thank you for using our program and for your kind Christmas wishes. I apologize for the inconvenience you're experiencing with missing program icons. This is a known issue that can occur in Windows, and there are several potential solutions. Here are some steps you can try to resolve the problem:
![2024-12-24 22 12 21](https://github.com/user-attachments/assets/9382cf56-9dbf-42a4-a9a2-2d16ad031a44)
![2024-12-24 22 10 21](https://github.com/user-attachments/assets/a3c893cd-3bcc-41c1-9c2e-f77286be9e5a)

**Comment by InsertX2k** [comment]
I apologize for any issues you may be having, in order to be able to provide you with the assistance you need I have to know all of the following information:
* Which version of Temp_Cleaner GUI are you running?
* Can you provide the MD5 or SHA512 hash of the installer you used to install Temp_Cleaner GUI on your computer using?
* What version of Windows are you running (Windows 10, 11, 7, 8.1, 8)?, also what release version of that OS are you running (e.g. 22H2, 24H2, 6.1.7601)?
* Which architecture is your OS running at? (x64, x86, arm64, arm)?
* How many RAM does your system have?
* Have you tried rebooting your computer to see if that fixes the problem?

According to the screenshots you sent I can see that the icons of "Adobe Acrobat" are missing, because they are stored in "%Windir%\Installer" directory, and by default the Windows Update cleanup does delete all files in that directory, the same behavior is reproducable with Windows' built-in disk cleanup tool and other PC cleaning software, your program shouldn't store it's icon in that directory, that might be a sign that it isn't installed correctly.

**Comment by InsertX2k** [comment, marked as answer]
This behavior also occurs with some applications that store their icons in `%windir%\Installer` folder, you can fix this issue just by changing the icon of the shortcut to be the main executable of the program.
[This one is a tutorial on how to find the executable of a program from the shortcut file of it (This is a hyperlink btw, you can click on it) (Not sponsored or affiliated)](https://www.howtogeek.com/688781/how-to-quickly-locate-a-programs-exe-file-on-windows-10/) *
*Please note that this link is NOT an official Temp_Cleaner GUI link nor the website is owned by me, You may find ads that promote PUPs (Potentially Unwanted Programs), please DO NOT click on those, also this website is not promoted by me nor do I get any sort of support from them, This link is just provided for convenience.

Also, Temp_Cleaner GUI warns you about it deleting the update files (which includes Windows Installer .msi files too), and Temp_Cleaner GUI cleans up this directory to ensure that Windows Updates downloaded are completely deleted and that the user can reboot their system without needing to wait for updates to be installed.

I hope that helps making the idea behind this issue clear to you.
